Nominatim 3.7

leonardehrenfried opened the pull request:
This is very much work in progress but I thought I put this up here nevertheless so that we can take a look at it together:

Things that need testing:

- [x] Updates
- [x] Postcode imports
- [x] Wikipedia imports

Fixes #197

philipkozeny commented on .github/workflows/ci.yml:
```diff
@@ -14,18 +14,21 @@ jobs:
       - name: Build docker image
         run: docker build . -t nominatim
-        working-directory: 3.6
+        working-directory: 3.7
```
do you think we can build for several versions? otherwise we would need to manually build / update older versions if there is a change.

leonardehrenfried replied:
I think that would be possible using a matrix build.

However, I'm snowed in with work in the next couple of weeks so if someone wants to take this to completion, please go ahead.

leonardehrenfried commented:
I think I'm done with the first version and would appreciate a review.

3.7 has separated the the project directory from the binary directory, introduced simpler way of configuring and added a new command line tool called `nominatim` replacing the previous salad of PHP commands.

leonardehrenfried commented:
By the way, since I download the tarball and don't clone the git repository anymore the country_grid file doesn't need to be separately downloaded.

philipkozeny commented on .github/workflows/ci.yml:
```diff
@@ -8,24 +8,38 @@ jobs:
+            --name nominatim \
             nominatim &
-          sleep 45
+          sleep 35
```
AFAIK 35 seconds is not enough, see: https://github.com/mediagis/nominatim-docker/runs/2507636802?check_suite_focus=true#step:4:360

If the download is not fast enough we may run into problems.

leonardehrenfried replied:
The script to check the response has a retry mechanism with an exponential backoff. It just retries for up to ~ a minute before giving up. In my view that is safe enough.

philipkozeny commented on 3.7/README.md:
```diff
@@ -0,0 +1,115 @@
+  - `POSTGRES_CHECKPOINT_TIMEOUT` (default: `10min`)
+  - `POSTGRES_CHECKPOINT_COMPLETITION_TARGET` (default: `0.9`)
+
+See https://nominatim.org/release-docs/3.7.0/admin/Installation/#tuning-the-postgresql-database for more details on those settings.
```
```suggestion
See https://nominatim.org/release-docs/3.7.1/admin/Installation/#tuning-the-postgresql-database for more details on those settings.
```

philipkozeny commented on 3.7/README.md:
```diff
@@ -0,0 +1,115 @@
+  - `full` : Default style that also includes points of interest.
+  - `extratags` : Like the full style but also adds most of the OSM tags into the extratags column.
+
+See https://nominatim.org/release-docs/3.7.0/admin/Import/#filtering-imported-data for more details on those styles.
```
```suggestion
See https://nominatim.org/release-docs/3.7.1/admin/Import/#filtering-imported-data for more details on those styles.
```

philipkozeny commented on 3.7/README.md:
```diff
@@ -0,0 +1,115 @@
+
+## Updating the database
+
+Full documentation for Nominatim update available [here](https://nominatim.org/release-docs/3.7.0/admin/Update/). For a list of other methods see the output of:
```
```suggestion
Full documentation for Nominatim update available [here](https://nominatim.org/release-docs/3.7.1/admin/Update/). For a list of other methods see the output of:
```

philipkozeny commented on 3.7/startapache.sh:
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+cp /data/local.php /app/src/build/settings/local.php
+/app/src/build/utils/setup.php --setup-website
```
is this still needed since you run ```refresh --website``` in ```start.sh```?

leonardehrenfried replied:
Good question. Does "website" refer to the PHP API or die UI with a map view? In other words: if I remove it and the test pass, are we sure that we haven't broken anything?

mtmail replied:
PHP API. https://github.com/osm-search/nominatim-ui is a separate project now which you'd either place into a subdirecty in Apache or setup a separate webserver. It's no longer part of the Nominatim install instructions.

leonardehrenfried replied:
Ok, thanks for the explanation. I removed the second call and all tests pass, so it wasn't needed.

philipkozeny commented:
Awesome work @leonardehrenfried ! Thanks a lot.

One last thing for discussion before we merge. With 3.7 the Nominatim team introduced a migration tool (https://nominatim.org/release-docs/latest/admin/Migration/#database-migrations). Do we want to integrate a ```sudo -u nominatim nominatim admin --migrate``` in our start.sh or just document it in the ```README.md``` (@mtmail any thoughts on that?)

leonardehrenfried commented:
Personally, I re-import quite frequently so I probably wouldn't use such a tool but others might.

My gut feeling is that since the migration can take a while I wouldn't do it automatically. 

I'm interested in hearing other people's feedback.

mtmail commented:
I'd keep the migration separate. It's currently not possible to check the Nominatim version. Some migration steps can indeed take hours.
